pkg/eventlog/recorder.go

```diff
@@ -102,6 +108,12 @@ func NewRecorder(
 			i.eventC = make(chan EventRecord, v)
 		case fileSplitterOpt:
 			i.newDests = v
+		case eventFilterOpt:
+			oldFilter := i.filter
+			i.filter = func(e *eventpb.Event) bool {
+				// Apply the given filter on top of the existing one.
+				return oldFilter(e) && v(e)
```

Comment:
I see what you mean. I could do it that way, but I think that the current approach is a bit more intuitive. You probably assume that the filters can be added dynamically, which is currently not the case. All the filter options are specified at instantiation of the recorder: `NewRecorder(... opt1, opt2, opt3, ...)` and some of these options are filter options. This loop just cycles through those option arguments. If we replaced the old filter by the new one, the semantics would be "ignore all filter options except for the last one". So I felt that it would be more convenient to "apply" all the specified filters, which translates to an AND. Of course the programer could achieve the same by specifying one combined filter. This is just sugar.